modules/phase_field/include/materials/OPInterfaceBarrierMaterial.h

```diff
@@ -0,0 +1,49 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef OPINTERFACEBARRIER_H
+#define OPINTERFACEBARRIER_H
+
+#include "Material.h"
+#include "DerivativeMaterialInterface.h"
+
+// Forward Declarations
+class OPInterfaceBarrierMaterial;
+
+template<>
+InputParameters validParams<OPInterfaceBarrierMaterial>();
+
+/**
+ * OPInterfaceBarrierMaterial is a constraint kernel that acts on all of the eta_i variables to
```

Comment:
No, it is not a kernel. It is a Free Energy penalty contribution (and a Material)